tests/unit/src/test/scala/tests/TypeDefinitionLspSuite.scala

```diff
@@ -0,0 +1,143 @@
+    extends BaseLspSuite("typeDefinition")
+    with TestHovers {
+  override def serverConfig: MetalsServerConfig =
+    super.serverConfig.copy(
+      statistics = new StatisticsConfig("diagnostics")
+    )
+
+  check("int")(
+    """
+      |object Main {
+      |  val ts@@t: Int = 2
+      |}""".stripMargin,
+    ".metals/readonly/scala/Int.scala [25:21 -> 25:24]"
+  )
+
+  check("basic")(
+    """
+      |package a
+      |
+      |class Main(i: Int) {}
+      |
+      |object Main extends App {
+      |  val te@@st = new Main(1)
+      |}
+        """.stripMargin,
+    "a/src/main/scala/a/Main.scala [3:6 -> 3:9]"
+  )
+
+  check("method")(
+    """
+      |package a
+      |
+      |class Main(i: Int) {}
+      |
+      |object Main extends App {
+      |  def foo(mn: Main): Unit = {
+      |     println(m@@n)
+      |  }
+      |}
+        """.stripMargin,
+    "a/src/main/scala/a/Main.scala [3:6 -> 3:9]"
+  )
+
+  check("method-definition")(
+    """
+      |package a
+      |class Main(i: Int) {}
+      |
+      |object Main extends App {
+      |  def foo(mn: Main): Unit = {
+      |     println(mn)
+      |  }
+      |
+      |  fo@@o(new Main(1))
+      |}
+        """.stripMargin,
+    "a/src/main/scala/a/Main.scala [5:6 -> 5:8]"
+  )
+
+  check("named-parameter")(
+    """
+      |case class CClass(str: String) {}
+      |
+      |object Main {
+      |  def tst(par: CClass): Unit = {}
+      |
+      |  tst(p@@ar = CClass("dads"))
+      |}""".stripMargin,
+    "a/src/main/scala/a/Main.scala [4:6 -> 4:8]"
+  )
+
+  check("pattern-match")(
+    """
+      |case class CClass(str: String) {}
+      |
+      |object Main {
+      |  CClass("test") match {
+      |    case CClass(st@@r) =>
+      |       println(str)
+      |    case _ =>
+      |  }
+      |}""".stripMargin,
+    "a/src/main/scala/a/Main.scala [3:16 -> 3:18]"
+  )
+
+  def check(name: String)(testCase: String, expected: String): Unit = {
+    val base = testCase.replaceAll("@@", "")
+    testAsync(name) {
+      cleanWorkspace()
+      for {
+        _ <- server.initialize(
+          s"""/metals.json
+             |{"a":{}}
+             |/a/src/main/scala/a/Main.scala
+             |$base
+      """.stripMargin
+        )
+        _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+        _ <- server.assertTypeDefinition(
+          "a/src/main/scala/a/Main.scala",
+          testCase,
+          locationFromString(expected)
+        )
+      } yield ()
+    }
+  }
+
+  def locationFromString(str: String): l.Location = {
+
+    val regex =
+      """\s*([^\s]+)\s* \[([0-9]+):([0-9]+)\s* ->\s* ([0-9]+):([0-9]+)\]\s*""".r
+
+    str match {
+      case regex(uri, startL, startC, endL, endC) =>
+        createLocation(
+          workspace.toURI.resolve(uri).toString.replaceFirst("/", "///"),
+          (startL.toInt, startC.toInt),
+          (endL.toInt, endC.toInt)
+        )
+      case _ =>
+        fail("Incorrect location string format")
```

Comment:
print the malformed string, so the user doesn't have to guess how it looked like

Reply:
Changed.